apps/desktop-wallet/src/pages/UnlockedWallet/OverviewPage/AssetsList.tsx

```diff
@@ -246,14 +249,18 @@ const NFTsList = ({ className, addressHashes, isExpanded, onExpand }: AssetsList
         ) : (
           <NFTList role="row" tabIndex={isExpanded ? 0 : -1}>
             {nfts.map((nft) => (
-              <NFTCard key={nft.id} nft={nft} />
+              <NFTCard key={nft.id} nft={nft} onClick={() => setSelectedNFTId(nft.id)} />
             ))}
             {nfts.length === 0 && <PlaceholderText>{t('No NFTs found.')}</PlaceholderText>}
           </NFTList>
         )}
       </motion.div>
 
       {!isExpanded && nfts.length > 4 && onExpand && <ExpandRow onClick={onExpand} />}
+
+      <ModalPortal>
+        {selectedNFTId && <NFTDetailsModal NFTId={selectedNFTId} onClose={() => setSelectedNFTId(undefined)} />}
+      </ModalPortal>
```

Comment:
This could be rendered inside the NFTCard. This way the whole asset list won't rerender when clicking on a card. Similar approach has been followed in the mobile wallet in the nft thumbnail.

Reply:
Didn't think about the whole list being rerendered but I trust your past work on the subject! Will do.

Reply:
Yup, can confirm that the modal opening is way faster when the modal is in the AssetsList.